### ISSUE TITLE
Query: data corruption for queries with set result operators (Concat/Union etc) and collection navigations

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFramework#8524]
query:

```
context.Gears.Select(g => g.Weapons.Concat(g.Weapons)).ToList()
```

produces bad QM:

```
from Gear g in DbSet<Gear>
select 
    (from Weapon <generated>_1 in DbSet<Weapon>
    where Property([g], "FullName") ?= Property([<generated>_1], "OwnerFullName")
    where Property([g], "FullName") ?= Property([<generated>_1], "OwnerFullName")
    select [<generated>_1]).Concat({value(Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable`1[Microsoft.EntityFrameworkCore.Specification.Tests.TestModels.GearsOfWarModel.Weapon])})
```

collection expansion predicate is applied twice to the outer query source, but is not applied to the inner one at all. This produces incorrect results (since there is no filter on the inner query source at all)

**Comment by maumar** [comment]
could be related to #8186

**Comment by maumar** [comment]
fixed in e1d7b1659d5f67da9db24a31038e0ca14a8c27d5
